Framework/PythonInterface/plugins/algorithms/LoadGudrunOutput.py

```diff
@@ -19,16 +19,37 @@ def summary(self):
 
     def PyInit(self):
         self.declareProperty(FileProperty(name='InputFile', defaultValue='', action=FileAction.Load,
-                                          extensions=['dcs01', 'mdcs01', 'mint01', 'mdor01', 'mgor01']))
-        self.declareProperty(WorkspaceProperty(name='OutputWorkspace', defaultValue='', direction=Direction.Output))
+                                          extensions=[".dcs01",
+                                                      ".mdsc01",
+                                                      ".mint01",
+                                                      ".mdor01",
+                                                      ".mgor01"]),
+                             doc="Gudrun output file to be loaded.")
+        self.declareProperty(WorkspaceProperty(name='OutputWorkspace', defaultValue='None',
```

Comment:
Have you tried marking this as optional, with default value empty? I'm a bit hesitant about automatic generation of names; while it makes sense when running from dialog, it gets confusing when running from python scripts. If the output is mandatory it makes more sense to me to force typing a name for it.

Reply:
Much cleaner solution - thanks